backend/app/store/engine/bolt.go

```diff
@@ -617,6 +645,80 @@ func (b *BoltDB) flagBucket(tx *bolt.Tx, flag Flag) (bkt *bolt.Bucket, err error
 	return bkt, nil
 }
 
+// checkUserDetail returns requested userDetail
+func (b *BoltDB) checkUserDetail(req UserDetailRequest) (val string) {
+	key := req.UserID
+	if key == "" {
+		return "" // return nothing in case UserID is not set
+	}
+
+	bdb, err := b.db(req.Locator.SiteID)
+	if err != nil {
+		return ""
+	}
+
+	_ = bdb.View(func(tx *bolt.Tx) error {
+		var bucket *bolt.Bucket
+		if bucket, err = b.userDetailsBucket(tx, req.Detail); err != nil {
+			return err
+		}
+		v := bucket.Get([]byte(key))
+		val = string(v)
+		return nil
+	})
+
+	return val
+}
+
+// checkUserDetail sets requested userDetail
+func (b *BoltDB) setUserDetail(req UserDetailRequest) (res string, err error) {
+	key := req.UserID
+	if key == "" {
+		return "", errors.New("UserID is not set")
+	}
+	if req.Delete && req.Update != "" {
+		return "", errors.New("Both Delete and Update are set, pick one")
+	}
+
+	bdb, e := b.db(req.Locator.SiteID)
+	if e != nil {
+		return "", e
+	}
+
+	err = bdb.Update(func(tx *bolt.Tx) error {
+		var bucket *bolt.Bucket
+		if bucket, err = b.userDetailsBucket(tx, req.Detail); err != nil {
+			return err
+		}
+		switch {
+		case req.Delete:
+			if e = bucket.Delete([]byte(key)); e != nil {
+				return errors.Wrapf(e, "failed to clean flag %s for %s", req.Detail, req.Locator.URL)
+			}
+			res = ""
+		case req.Update != "":
+			if e = bucket.Put([]byte(key), []byte(req.Update)); e != nil {
+				return errors.Wrapf(e, "failed to set detail %s for %s", req.Detail, req.Locator.URL)
+			}
+			res = req.Update
+		}
+		return nil
+	})
+
+	return res, err
+}
+
+func (b *BoltDB) userDetailsBucket(tx *bolt.Tx, userDetail UserDetail) (bkt *bolt.Bucket, err error) {
```

Comment:
function name as well as parameter confusing. Took me time to figure how userDetail in 3 places mean 3 different things in one function. 

This gets a bucket for subset of userDetails and if you can't figure better name you need at lest to comment what it does